GitHub-Campus-Experts-Unilag/DevzenFrontend
# feat jwt-debugger into signin page

### Title: Add Google and GitHub OAuth Social Login

---

### Description

#### Changes
- Adds jwt-debugger for authentication.
- Refactors the authentication logic to support OAuth-based sign-in.

#### Features
- This feature improves the user experience by offering alternative login options (Google and GitHub OAuth).
- It allows users to authenticate using their existing Google or GitHub accounts.

#### How does this impact the application?
- Sign up using username, email, and password.
- Adds two new social authentication options.
- UI updates to accommodate the new buttons for OAuth sign-in.

#### Key Changes:
-Added jwt-decode package to handle sign-up.
- Added Google and GitHub OAuth integration in `SignIn.js`.
- Created utility functions for handling OAuth in `auth.js`.
- Updated UI to include new Google and GitHub login buttons.


#### Dependencies
- Added `react-oauth/google` for Google OAuth support.
- Updated `react-icons` for GitHub icon integration.

#### Testing Instructions:
1. Navigate to the sign-in page.
2. Click the "Continue with Google" button and verify that the OAuth flow completes successfully.
3. Click the "Continue with GitHub" button and verify that the OAuth flow completes successfully.
4. Ensure that both OAuth methods properly decode and handle user tokens.
